src/edit.c

```diff
@@ -2455,8 +2455,8 @@ stop_insert(
     int added = inserted.string == NULL ? 0 : (int)inserted.length - new_insert_skip;
     if (did_restart_edit == 0 || added > 0)
     {
-	vim_free(last_insert);
-	last_insert = inserted.string;
+	vim_free(last_insert.string);
+	last_insert = inserted;			    // structure copy
```

Comment:
I don't think code that isn't complex needs comments.
```suggestion
	last_insert = inserted;
```